server/src/main/java/org/elasticsearch/index/mapper/GeoShapeIndexer.java

```diff
@@ -261,7 +262,15 @@ public Void visit(Polygon polygon) {
 
         @Override
         public Void visit(Rectangle r) {
-            addFields(LatLonShape.createIndexableFields(name, GeoShapeUtils.toLucenePolygon(r)));
+            if (r.getMinLon() > r.getMaxLon()) {
+                Rectangle left = new Rectangle(r.getMaxX(), GeoUtils.MAX_LON, r.getMaxY(), r.getMinY());
+                addFields(LatLonShape.createIndexableFields(name, GeoShapeUtils.toLucenePolygon(left)));
+                Rectangle right = new Rectangle(GeoUtils.MIN_LON, r.getMaxLon(), r.getMaxY(), r.getMinY());
```

Comment:
Maybe be better to stick with X, Y or Lat, Lon for clarity.

Reply:
True! I will go for Lat / Lon